assignment4.py

```diff
@@ -0,0 +1,179 @@
+        self.uf_dict[idx] = UnionFindElement(idx)
+
+    def union(self, i, j):
+        """
+        If elements are from different sets, performs union and returns 1.
+        Returns 0 if both elements are members of the same set.
+        """
+        root_i = self.uf_dict[self.find(i)]
+        root_j = self.uf_dict[self.find(j)]
+        if root_i == root_j:
+            return 0
+        if root_i.rank > root_j.rank:
+            root_j.parent = root_i.parent
+        elif root_i.rank < root_j.rank:
+            root_i.parent = root_j.parent
+        else:
+            root_i.parent = root_j.parent
+            root_j.rank += 1
+        return 1
+
+    def find(self, x):
+        """ Returns a key of an element in a union-find dict """
+        while self.uf_dict[x].parent != x:
+            x = self.uf_dict[x].parent
+        return x
+
+
+class IslandMap():
+
+    def __init__(self, nrows, ncols, isl_array):
+        self.nrows = nrows
+        self.ncols = ncols
+        self.islands = isl_array
+
+    def isIsland(self, i, j):
+        if (i < 0) or (i >= self.nrows):
+            return False
+        if (j < 0) or (j >= self.ncols):
+            return False
+        return self.islands[i][j]
+
+    def countIslands(self):
+        """
+        Counts islands using a union-find data structure in an iterative way.
+        Returns an integer -- number of islands on a map
+        """
+        isls_uf = UnionFind()
+        counter = 0
+        for i in range(self.nrows):
+            for j in range(self.ncols):
+                if self.isIsland(i, j):
+                    isls_uf.make_set(i * self.ncols + j)
+                    counter += 1
+                    if self.isIsland(i-1, j):
+                        if isls_uf.union((i-1) * self.ncols + j, i * self.ncols + j):
+                            counter -= 1
+                    if self.isIsland(i, j-1):
+                        if isls_uf.union(i * self.ncols + j-1, i * self.ncols + j):
+                            counter -= 1
+        return counter
+
+    def countIslandsDFS(self, i, j, visited):
+        """ Auxiliary recursive function for countIslandsRecursive """
+        visited.add(i * self.ncols + j)
+        for d in ([0, 1], [1, 0], [0, -1], [-1, 0]):
+            new_i = i+d[0]
+            new_j = j+d[1]
+            if ((new_i * self.ncols + new_j) not in visited) and self.isIsland(new_i, new_j):
+                self.countIslandsDFS(new_i, new_j, visited)
+        return
+
+    def countIslandsRecursive(self):
+        """ 
+        Counts islands using Depth-First search in a recursive way.
+        """
+        visited = set()
+        counter = 0
+        for i in range(self.nrows):
+            for j in range(self.ncols):
+                if ((i * self.ncols + j) not in visited) and self.isIsland(i, j):
+                    counter += 1
+                    self.countIslandsDFS(i, j, visited)
+        return counter
+
+
+T = True
```

Comment:
0 and 1 evaluate as False and True in boolean checks, so maybe you could use those.